### PR TITLE
fix blocking of syslog writer (tcp) or use udp

### PR DESCRIPTION
**Pull request by xjewer**
Cause `log/syslog` is no longer supported, it was copied to this library and methods was added to set up connection DeadLine's

plus fix some typos

### CODE REVIEW

File: syslog_writer.go [hunk 1]
```diff
@@ -2,47 +2,75 @@ package golog
...
 
-// NewSyslogWriter return new SyslogWriter instance with cuncurrent wirter to syslog
-func NewSyslogWriter(addr, tag string, timeout int) *SyslogWriter {
-	s := &SyslogWriter{}
-	w, err := syslog.Dial("tcp", addr, syslog.LOG_USER, tag)
-
+// NewSyslogWriter return new SyslogWriter instance with concurrent writer to syslog
+func NewSyslogWriter(network, addr, tag string, timeout int) *SyslogWriter {
+	s := &SyslogWriter{
+		timeout: time.Second * time.Duration(timeout),
```
Comment (chapsuk) [inline syslog_writer.go]: It would be nice to pass timeout as `time.Duration` instead of `int`

File: syslog_writer.go [hunk 2]
```diff
@@ -2,47 +2,75 @@ package golog
...
+	s.Writer = w
 	return s
 }
 
+// Write writes data to syslog
 func (w *SyslogWriter) Write(p []byte) (int, error) {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	if sl, ok := w.Writer.(*syslog.Writer); ok {
+		err := sl.SetWriteDeadLine(time.Now().Add(w.timeout))
```
Comment (chapsuk) [inline syslog_writer.go]: It is enough to set the DeadLine timeout once after create `syslog.Writer`

File: syslog/syslog_unix.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+func unixSyslog() (conn serverConn, err error) {
+	logTypes := []string{"unixgram", "unix"}
+	logPaths := []string{"/dev/log", "/var/run/syslog", "/var/run/log"}
+	for _, network := range logTypes {
+		for _, path := range logPaths {
+			conn, err := net.Dial(network, path)
+			if err != nil {
+				continue
+			} else {
```
Comment (chapsuk) [inline syslog/syslog_unix.go]: `else` not needed

Comment (xjewer) [inline syslog/syslog_unix.go]: `bike is not mine` 😂

File: syslog_writer.go (hunk 2 above)
Comment (xjewer) [inline syslog_writer.go]: no, It is not, cause you set specific time, not time.Duration

File: syslog/syslog_unix.go (hunk 3 above)
Comment (chapsuk) [inline syslog/syslog_unix.go]: ok, but fix it please

Comment (xjewer) [inline syslog/syslog_unix.go]: 👌

**Pull request merged**